ichnaea/locate/tests/test_provider.py

```diff
@@ -505,3 +512,146 @@ def test_geoip_country(self):
         self.assertEqual(type(location), Country)
         self.assertEqual(location.country_code, bhutan['country_code'])
         self.assertEqual(location.country_name, bhutan['country_name'])
+
+
+class TestFallbackProvider(ProviderTest):
+
+    TestProvider = FallbackProvider
+
+    def setUp(self):
+        super(TestFallbackProvider, self).setUp()
+
+        self.provider.api_key.allow_fallback = True
+
+        requests_patcher = mock.patch('ichnaea.locate.provider.requests.post')
+        self.mock_requests = requests_patcher.start()
+        self.addCleanup(requests_patcher.stop)
```

Comment:
For requests mocking, we already use the extra helper library mock_requests (https://requests-mock.readthedocs.org/en/latest/). There's a couple examples over here: https://github.com/mozilla/ichnaea/blob/master/ichnaea/data/tests/test_export.py#L139